server/src/main/java/org/elasticsearch/index/engine/CombinedDeletionPolicy.java

```diff
@@ -109,6 +130,14 @@ private void updateRetentionPolicy() throws IOException {
             Long.parseLong(safeCommit.getUserData().get(SequenceNumbers.LOCAL_CHECKPOINT_KEY)));
     }
 
+    protected int getDocCountOfCommit(IndexCommit indexCommit) throws IOException {
+        return SegmentInfos.readCommit(indexCommit.getDirectory(), indexCommit.getSegmentsFileName()).totalMaxDoc();
```

Comment:
Nested docs are treated as full Lucene docs using this metric whereas they do not receive a separate sequence number.

Reply:
We discussed this and decided to proceed as-is despite this issue with nested docs. We could not find a sufficiently cheap way to count the docs in the index accounting for nested docs. The behaviour in this PR is never worse than it was before, and is sometimes better.